pkg/util/overridemanager/overridemanager.go

```diff
@@ -190,7 +190,20 @@ func (o *overrideManagerImpl) getOverridersFromOverridePolicies(policies []Gener
 			resourceMatchingPolicies = append(resourceMatchingPolicies, policy)
 		}
 	}
-
+	sort.SliceStable(resourceMatchingPolicies, func(i, j int) bool {
+		implicitPriorityI := util.ResourceMatchSelectorsPriority(resource, resourceMatchingPolicies[i].GetOverrideSpec().ResourceSelectors...)
+		if len(resourceMatchingPolicies[i].GetOverrideSpec().ResourceSelectors) == 0 {
+			implicitPriorityI = util.PriorityMatchAll
+		}
+		implicitPriorityJ := util.ResourceMatchSelectorsPriority(resource, resourceMatchingPolicies[j].GetOverrideSpec().ResourceSelectors...)
+		if len(resourceMatchingPolicies[j].GetOverrideSpec().ResourceSelectors) == 0 {
```

Comment:
How about moving this logic to function `ResourceMatchSelectorsPriority`?

Reply:
> How about moving this logic to function ResourceMatchSelectorsPriority?

I think it may have an impact on [this](https://github.com/karmada-io/karmada/blob/master/pkg/detector/compare.go#L15-L23)

Reply:
I check the logic, the answer seems no.  what do you think? @XiShanYongYe-Chang

Reply:
> How about moving this logic to function ResourceMatchSelectorsPriority?
+1
It can also be said that the `ResourceMatchSelectorsPriority` calculated the OP priority incorrectly. When `resourceSelectors` is nil, the implicit priority should be 1.

Reply:
ok

Reply:
> How about moving this logic to function ResourceMatchSelectorsPriority?

I'm confused, I think it's more reasonable for the function `ResourceMatchSelectorsPriority ` to default return to `MisMatch` when selectors is nil.

Reply:
I still prefer to move the logic inside the function `ResourceMatchSelectorsPriority `, unit-test needs to be modified.

Reply:
@jwcesign 
https://github.com/karmada-io/karmada/pull/2609#discussion_r1034237730